iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java

```diff
@@ -66,31 +67,33 @@ String processHelmTemplate(String filename, String source, InputFileContext inpu
 
     // TODO: better support of Helm project structure
     var sourceWithComments = addLineComments(source);
-    var valuesFile = findValuesFile(inputFileContext);
-    var valuesFileContent = validateAndReadValuesFile(valuesFile, inputFileContext.inputFile);
-    // TODO: SONARIAC-1225 Pass all files of the Helm project directory to HelmEvaluator
-    var templateDependencies = Map.of(
-      "values.yaml", valuesFileContent);
-    return evaluateHelmTemplate(filename, inputFileContext.inputFile, sourceWithComments, templateDependencies);
+    Map<String, InputFile> additionalFiles = retrieveFilesInHelmProject(inputFileContext);
+    var fileContents = validateAndReadFiles(inputFileContext.inputFile, additionalFiles);
+    return evaluateHelmTemplate(filename, inputFileContext.inputFile, sourceWithComments, fileContents);
   }
 
-  private static String validateAndReadValuesFile(@Nullable InputFile valuesFile, InputFile inputFile) {
-    if (valuesFile == null) {
+  private static Map<String, String> validateAndReadFiles(InputFile inputFile, Map<String, InputFile> files) {
+    if (files.get("values.yaml") == null && files.get("values.yml") == null) {
```

Comment:
Why not `contains`?

Reply:
I previously used the result and forget to change it back. Good catch!